### PR TITLE
😱😎🙂💀😈

### PR DESCRIPTION
**Pull request by IRacle1**
## Description
**Describe the changes** 
1. Fixes `DoorVariant::RegisterRooms` patch, that prefix condition was created by me earlier to prevent `DoorVariant::Rooms` being broken(by basegame code rooms will be empty, when register rooms more than once), but somehow it was deleted😭 (I discovered that bug for MER doors, but Im sure it can affect regular doors(al least it was before))
2. Added new overload for `SendFakeSyncVars`, for fixing sending `null` syncvars (for example sending `null` player custom info)

**What is the current behavior?** (You can also link to an open issue here)
rooms are not registered
syncvars throw nre
😭

**What is the new behavior?** (if this is a feature change)
rooms are registered
syncvars dont throw nre
😎

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No🥲

**Other information**:
Other information
Error that lead to break syncvars, was caused by `value.GetType`, when value is `null`, i created type-template method so type is not taken directly from the object, and you can send, for example, `null` strings
<br />

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [x] I have checked no IL patching errors in the console
image

### Other
- [ ] Still requires more testing

### CODE REVIEW

File: EXILED/Exiled.Events/Patches/Generic/DoorList.cs [hunk 1]
```diff
@@ -30,6 +30,11 @@ namespace Exiled.Events.Patches.Generic
     [HarmonyPatch(typeof(DoorVariant), nameof(DoorVariant.RegisterRooms))]
     internal class DoorList
     {
+        private static bool Prefix(DoorVariant __instance)
+        {
+            return __instance.Rooms == null;
+        }
```
Comment (louis1706) [inline EXILED/Exiled.Events/Patches/Generic/DoorList.cs:36]: Can you il code :3333

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Review by Misfiy — APPROVED** [review]

File: EXILED/Exiled.Events/Patches/Generic/DoorList.cs (hunk 1 above)
Comment (IRacle1) [reply EXILED/Exiled.Events/Patches/Generic/DoorList.cs]: yep

File: EXILED/Exiled.Events/Patches/Generic/DoorList.cs [hunk 2]
```diff
@@ -30,14 +30,46 @@ namespace Exiled.Events.Patches.Generic
...
     {
-        private static void Postfix(DoorVariant __instance)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            if (Door.DoorVariantToDoor.ContainsKey(__instance))
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label ret = generator.DefineLabel();
+
+            // new Generator(this)
```
Comment (louis1706) [inline EXILED/Exiled.Events/Patches/Generic/DoorList.cs:39]: I didn't know generator was a door :33

File: EXILED/Exiled.Events/Patches/Generic/DoorList.cs [hunk 3]
```diff
@@ -30,14 +30,46 @@ namespace Exiled.Events.Patches.Generic
...
+                    new(OpCodes.Brfalse_S, ret),
+                });
+
+            newInstructions.InsertRange(
+                newInstructions.Count - 1,
+                new CodeInstruction[]
+                {
+                    new(OpCodes.Ldarg_0),
+                    new(OpCodes.Call, Method(typeof(DoorList), nameof(DoorList.InitDoor))),
+                });
```
Comment (louis1706) [inline EXILED/Exiled.Events/Patches/Generic/DoorList.cs:55]: More comment on what is happening there please :3

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: EXILED/Exiled.Events/Patches/Generic/DoorList.cs (hunk 2 above)
Comment (IRacle1) [reply EXILED/Exiled.Events/Patches/Generic/DoorList.cs:39]: 😭😭😭

**Review by VALERA771 — APPROVED** [review]

**Pull request merged**